src/fsharp/FSharp.Core/prim-types.fs

```diff
@@ -4790,7 +4790,7 @@ namespace Microsoft.FSharp.Core
                 System.Threading.Monitor.Exit(lockobj)
 #else
         [<CompiledName("Lock")>]
-        let inline lock (lockObject : 'T when 'T : not struct) action  = 
```

Comment:
BTW could you also remove the dead `#if` clause above? `Monitor.Enter(object, ref bool)` has been available since .NET 4.0.

Reply:
I don't mind changing that as well but is that ok as part of the same PR?